imports/plugins/core/catalog/server/methods/catalog.js

```diff
@@ -937,6 +937,8 @@ Meteor.methods({
       return numFlaggedAsDeleted;
     }
 
+    Hooks.Events.run("afterRemoveCatalogProduct", this.userId, product);
```

Comment:
Lets add a comment on this hooks, to make clear that it is updating the product in the `Catalog` collection.